onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -897,6 +898,55 @@ def _infer_Concat(self, node):  # noqa: N802
             )
         )
 
+    def _infer_stft(self, node):
+        # ref: https://github.com/onnx/onnx/blob/main/onnx/defs/math/defs.cc#L3471
+        one_sided = get_attribute(node, "onesided", 1)
+        signal_shape = self._get_shape(node, 0)
+        signal_dim = self._get_shape_rank(node, 0)
+
+        # The frame step is a required input.
+        # Its value is needed to compute the number output nDFTs, so return early is missing.
+        frame_step = self._get_int_or_float_values(node)[1]
+        if frame_step is None:
+            return
+        window_shape = self._get_shape(node, 2)
+
+        dft_size = -1
+        # If we cannot read the frame_length, we cannot infer shape but frame_length is optional
+        # return...
+        frame_length = self._get_int_or_float_values(node)[3]
+        if frame_length is None:
+            return
+
+        if window_shape is None and frame_length is None:
+            # STFT expects to have at least one of these inputs set: [window, frame_length],
+            # but they may not be available at shape inference time
+            return
+        elif window_shape is not None and frame_length is not None:
+            dft_size = frame_length
+        elif frame_length is not None:
+            dft_size = frame_length
+
+        if one_sided:
+            dft_unique_bins = (dft_size >> 1) + 1
```

Comment:
Bit shifting is error prone, consider

```suggestion
            dft_unique_bins = dft_size // 2 + 1
```

?